clang/lib/Sema/Metafunctions.cpp

```diff
@@ -2007,7 +2007,15 @@ bool type_of(APValue &Result, Sema &S, EvalFn Evaluator, DiagFn Diagnoser,
 
   switch (RV.getReflectionKind()) {
   case ReflectionKind::Null:
-  case ReflectionKind::Type:
+  case ReflectionKind::Type: {
+    QualType currentType = RV.getTypeOfReflectedResult(S.Context);
+    bool UnwrapAlises = isa<UsingType>(currentType) ||
+                        isa<TypedefType>(currentType) ||
+                        isa<TemplateSpecializationType>(currentType);
```

Comment:
Out of curiosity, what happens if we just pass `/*UnwrapAliases=*/true` unconditionally? Are we avoiding an undesirable behavior with this check, or is this an optimization? If the later, I'd prefer passing unconditional `true`.

(also for `return_type_of` below)

Reply:
I updated to unconditional true and it looks fine to me. I updated some tests in libcxx to reflect the new behavior.